Error while checking build/O2/o2-dataflow for 58cc6a7a31af76d8ab23f84ecab29affb03a0333 at 2022-08-31 17:54:
```
## sw/BUILD/O2-latest/log
100% tests passed, 0 tests failed out of 421
 95/106 Test #109: test_Framework_test_SingleDataSource ....................***Failed    3.09 sec
[ERROR] pid 24808 (internal-dpl-injected-dummy-sink) crashed with 1
[ERROR] SEVERE: Device internal-dpl-injected-dummy-sink (24808) returned with 1
99% tests passed, 1 tests failed out of 101
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/9734/58cc6a7a31af76d8ab23f84ecab29affb03a0333/build_O2_o2-dataflow/pretty.html).

--- a/Detectors/ITSMFT/ITS/workflow/include/ITSWorkflow/DCSParserSpec.h
+++ b/Detectors/ITSMFT/ITS/workflow/include/ITSWorkflow/DCSParserSpec.h
@@ -63,7 +63,7 @@ class ITSDCSParser : public Task
   void saveToOutput();
   void resetMemory();
   void pushToCCDB(ProcessingContext&);
-  bool updatePosition(unsigned int&, unsigned int&, const std::string&, const char*, const std::string&, bool ignoreNpos=false);
+  bool updatePosition(unsigned int&, unsigned int&, const std::string&, const char*, const std::string&, bool ignoreNpos = false);
   void updateAndCheck(int&, const int);
   void updateAndCheck(short int&, const short int);
   void writeChipInfo(o2::dcs::DCSconfigObject_t&, const std::string&, const unsigned short int);

--- a/Detectors/ITSMFT/ITS/workflow/include/ITSWorkflow/DCSParserSpec.h
+++ b/Detectors/ITSMFT/ITS/workflow/include/ITSWorkflow/DCSParserSpec.h
@@ -63,6 +63,7 @@ class ITSDCSParser : public Task
   void saveToOutput();
   void resetMemory();
   void pushToCCDB(ProcessingContext&);
+  bool updatePosition(unsigned int&, unsigned int&, const std::string&, const char*, const std::string&, bool ignoreNpos=false);
   void updateAndCheck(int&, const int);
   void updateAndCheck(short int&, const short int);
   void writeChipInfo(o2::dcs::DCSconfigObject_t&, const std::string&, const unsigned short int);

--- a/Detectors/ITSMFT/ITS/workflow/src/DCSParserSpec.cxx
+++ b/Detectors/ITSMFT/ITS/workflow/src/DCSParserSpec.cxx
@@ -175,8 +175,8 @@ void ITSDCSParser::updateMemoryFromInputString(const std::string& inString)
 //////////////////////////////////////////////////////////////////////////////
 // Update pos and npos and check for validity. Return false if there is error
 bool ITSDCSParser::updatePosition(unsigned int& pos, unsigned int& npos,
-  const std::string& delimiter, const char* word,
-  const std::string& inString, bool ignoreNpos/*=false*/)
+                                  const std::string& delimiter, const char* word,
+                                  const std::string& inString, bool ignoreNpos /*=false*/)
 {
   pos += npos + delimiter.length() + std::string(word).length();
   if (!ignoreNpos) {

--- a/Detectors/ITSMFT/ITS/workflow/src/DCSParserSpec.cxx
+++ b/Detectors/ITSMFT/ITS/workflow/src/DCSParserSpec.cxx
@@ -56,7 +56,8 @@ void ITSDCSParser::run(ProcessingContext& pc)
   // Check for DOS vs. Unix line ending
   std::string line_ending = "\n";
   size_t newline_pos = inStringConv.find(line_ending);
-  if (newline_pos && inStringConv[newline_pos - 1] == '\r') {
+  if (newline_pos && newline_pos != std::string::npos &&
+      inStringConv[newline_pos - 1] == '\r') {
     line_ending = "\r\n";
   }
 
@@ -90,41 +91,45 @@ void ITSDCSParser::updateMemoryFromInputString(const std::string& inString)
   const std::string delimiter = "|";
   unsigned int pos = 0;
   unsigned int npos = inString.find(delimiter);
+  if (npos == std::string::npos) {
+    LOG(error) << "Delimiter not found, possibly corrupted data!";
+    return;
+  }
 
   // First is the stave name
   this->mStaveName = inString.substr(pos, npos);
 
   // Next is the run number
-  std::string word = "RUN";
-  pos += npos + delimiter.length() + word.length();
-  npos = inString.find(delimiter, pos) - pos;
+  if (!this->updatePosition(pos, npos, delimiter, "RUN", inString)) {
+    return;
+  }
   this->updateAndCheck(this->mRunNumber, std::stoi(inString.substr(pos, npos)));
 
   // Next is the config
-  word = "CONFIG";
-  pos += npos + delimiter.length() + word.length();
-  npos = inString.find(delimiter, pos) - pos;
+  if (!this->updatePosition(pos, npos, delimiter, "CONFIG", inString)) {
+    return;
+  }
   this->updateAndCheck(this->mConfigVersion, std::stoi(inString.substr(pos, npos)));
 
   // Then it's the run type
-  word = "RUNTYPE";
-  pos += npos + delimiter.length() + word.length();
-  npos = inString.find(delimiter, pos) - pos;
+  if (!this->updatePosition(pos, npos, delimiter, "RUNTYPE", inString)) {
+    return;
+  }
   this->updateAndCheck(this->mRunType, std::stoi(inString.substr(pos, npos)));
 
   // Then there is a semicolon-delineated list of disabled chips
-  word = "DISABLED_CHIPS";
-  pos += npos + delimiter.length() + word.length();
-  npos = inString.find(delimiter, pos) - pos;
+  if (!this->updatePosition(pos, npos, delimiter, "DISABLED_CHIPS", inString)) {
+    return;
+  }
   std::string disabledChipsStr = inString.substr(pos, npos);
   if (disabledChipsStr.length()) {
     this->mDisabledChips = this->vectorizeStringListInt(disabledChipsStr, ";");
   }
 
   // Then there is a 2D list of masked double-columns
-  word = "MASKED_DCOLS";
-  pos += npos + delimiter.length() + word.length();
-  npos = inString.find(delimiter, pos) - pos;
+  if (!this->updatePosition(pos, npos, delimiter, "MASKED_DCOLS", inString)) {
+    return;
+  }
   std::string maskedDoubleColsStr = inString.substr(pos, npos);
   if (maskedDoubleColsStr.length()) {
     std::vector<std::string> chipVect = this->vectorizeStringList(maskedDoubleColsStr, ";");
@@ -135,8 +140,9 @@ void ITSDCSParser::updateMemoryFromInputString(const std::string& inString)
   }
 
   // Finally, there are double columns which are disabled at EOR
-  word = "DCOLS_EOR";
-  pos += npos + delimiter.length() + word.length();
+  if (!this->updatePosition(pos, npos, delimiter, "DCOLS_EOR", inString, true)) {
+    return;
+  }
   std::string doubleColsEORstr = inString.substr(pos);
   if (doubleColsEORstr.length()) {
     std::vector<std::string> bigVect = this->vectorizeStringList(doubleColsEORstr, "&");
@@ -164,6 +170,28 @@ void ITSDCSParser::updateMemoryFromInputString(const std::string& inString)
   }
 
   return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// Update pos and npos and check for validity. Return false if there is error
+bool ITSDCSParser::updatePosition(unsigned int& pos, unsigned int& npos,
+  const std::string& delimiter, const char* word,
+  const std::string& inString, bool ignoreNpos/*=false*/)
+{
+  pos += npos + delimiter.length() + std::string(word).length();
+  if (!ignoreNpos) {
+    npos = inString.find(delimiter, pos);
+
+    // Check that npos does not go out-of-bounds
+    if (npos == std::string::npos) {
+      LOG(error) << "Delimiter not found, possibly corrupted data!";
+      return false;
+    }
+
+    npos -= pos;
+  }
+
+  return true;
 }
 
 //////////////////////////////////////////////////////////////////////////////

--- a/Detectors/ITSMFT/ITS/workflow/src/DCSParserSpec.cxx
+++ b/Detectors/ITSMFT/ITS/workflow/src/DCSParserSpec.cxx
@@ -89,8 +89,8 @@ void ITSDCSParser::updateMemoryFromInputString(const std::string& inString)
 
   // Parse the individual parts of the string
   const std::string delimiter = "|";
-  unsigned int pos = 0;
-  unsigned int npos = inString.find(delimiter);
+  size_t pos = 0;
+  size_t npos = inString.find(delimiter);
   if (npos == std::string::npos) {
     LOG(error) << "Delimiter not found, possibly corrupted data!";
     return;
@@ -174,7 +174,7 @@ void ITSDCSParser::updateMemoryFromInputString(const std::string& inString)
 
 //////////////////////////////////////////////////////////////////////////////
 // Update pos and npos and check for validity. Return false if there is error
-bool ITSDCSParser::updatePosition(unsigned int& pos, unsigned int& npos,
+bool ITSDCSParser::updatePosition(size_t& pos, size_t& npos,
                                   const std::string& delimiter, const char* word,
                                   const std::string& inString, bool ignoreNpos /*=false*/)
 {
@@ -231,7 +231,7 @@ std::vector<std::string> ITSDCSParser::vectorizeStringList(
   const std::string& str, const std::string& delimiter)
 {
   std::vector<std::string> str_vect;
-  std::string::size_type prev_pos = 0, pos = 0;
+  size_t prev_pos = 0, pos = 0;
   while ((pos = str.find(delimiter, pos)) != std::string::npos) {
     std::string substr = str.substr(prev_pos, pos - prev_pos);
     if (substr.length()) {
@@ -252,7 +252,7 @@ std::vector<unsigned short int> ITSDCSParser::vectorizeStringListInt(
   const std::string& str, const std::string& delimiter)
 {
   std::vector<unsigned short int> int_vect;
-  std::string::size_type prev_pos = 0, pos = 0;
+  size_t prev_pos = 0, pos = 0;
   while ((pos = str.find(delimiter, pos)) != std::string::npos) {
     unsigned short int substr = std::stoi(str.substr(prev_pos, pos - prev_pos));
     int_vect.push_back(substr);

